pacman/game.py

```diff
@@ -33,15 +33,43 @@ def add_ghost(self, agent):
             raise OutsideOfLegalPath("Cannot add agent\
                 to invalid board position")
 
-    def play_game(self, game_steps):
+    def check_rules(self, pacman_old_node, pacman_new_node,
+                    ghost_old_node, ghost_new_node):
+        collision = False
+        if (pacman_new_node.position == ghost_new_node.position):
+            collision = True
+        if ((pacman_new_node.position == ghost_old_node.position) and (
+                pacman_old_node.position == ghost_new_node.position)):
+            collision = True
+        return collision
+
+    def play_game(self):
         game_reward = 0
-        for step in range(game_steps):
-            self.pac_man.move()
-            game_reward += self.pac_man.current_node.reward
-            self.pac_man.current_node.reward = 0
-            for ghost in self.ghosts:
-                ghost.move()
+        game_finished = False
+        while (not game_finished):
+            # Compute next moves
+            next_pacman_node = self.pacman.get_move()
+            next_ghost_nodes = [ghost.get_move() for ghost in self.ghosts]
+            current_ghost_nodes = [ghost.current_node for ghost in self.ghosts]
+
+            # Check if game finished
+            conflicts = [self.check_rules(self.pacman.current_node,
+                                          next_pacman_node, current_ghost_node,
+                                          next_ghost_node)
+                         for current_ghost_node, next_ghost_node
+                         in zip(current_ghost_nodes, next_ghost_nodes)]
```

Comment:
More readable to put it in a real loop.
Inline for loops should only be used with small readable commands.